src/EFCore/DbContext.cs

```diff
@@ -726,27 +726,24 @@ private void SetEntityState(InternalEntityEntry entry, EntityState entityState)
             }
         }
 
-        private async Task SetEntityStateAsync(
+        private Task SetEntityStateAsync(
             InternalEntityEntry entry,
             EntityState entityState,
             CancellationToken cancellationToken)
         {
             if (entry.EntityState == EntityState.Detached)
             {
-                await DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
+                return DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
```

Comment:
I may be misreading the code, but isn't the following possible?

```suggestion
        private Task SetEntityStateAsync(
            InternalEntityEntry entry,
            EntityState entityState,
            CancellationToken cancellationToken)
        => entry.EntityState == EntityState.Detached
            ? DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
                entry,
                entityState,
                forceStateWhenUnknownKey: true,
                cancellationToken: cancellationToken)
            : entry.SetEntityStateAsync(
                entityState,
                acceptChanges: true,
                forceStateWhenUnknownKey: entityState,
                cancellationToken: cancellationToken);
                cancellationToken: cancellationToken);
```